lib/yargs.ts

```diff
@@ -0,0 +1,1694 @@
+      completionCommand,
+      parseFn,
+      parseContext,
+      handlerFinishCommand
+    } = frozen)
+    options.configObjects = configObjects
+    usage.unfreeze()
+    validation.unfreeze()
+    command.unfreeze()
+  }
+
+  self.boolean = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('boolean', keys)
+    return self
+  }
+
+  self.array = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('array', keys)
+    return self
+  }
+
+  self.number = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('number', keys)
+    return self
+  }
+
+  self.normalize = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('normalize', keys)
+    return self
+  }
+
+  self.count = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('count', keys)
+    return self
+  }
+
+  self.string = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('string', keys)
+    return self
+  }
+
+  self.requiresArg = function (keys) {
+    // the 2nd paramter [number] in the argsert the assertion is mandatory
+    // as populateParserHintSingleValueDictionary recursively calls requiresArg
+    // with Nan as a 2nd parameter, although we ignore it
+    argsert('<array|string|object> [number]', [keys], arguments.length)
+    // If someone configures nargs at the same time as requiresArg,
+    // nargs should take precedent,
+    // see: https://github.com/yargs/yargs/pull/1572
+    // TODO: make this work with aliases, using a check similar to
+    // checkAllAliases() in yargs-parser.
+    if (typeof keys === 'string' && options.narg[keys]) {
+      return self
+    } else {
+      populateParserHintSingleValueDictionary(self.requiresArg, 'narg', keys, NaN)
+    }
+    return self
+  }
+
+  self.skipValidation = function (keys) {
+    argsert('<array|string>', [keys], arguments.length)
+    populateParserHintArray('skipValidation', keys)
+    return self
+  }
+
+  function populateParserHintArray<T extends KeyOf<Options, string[]>> (
+    type: T,
+    keys: string | string[]
+  ) {
+    keys = ([] as string[]).concat(keys)
+    keys.forEach((key) => {
+      key = sanitizeKey(key)
+      options[type].push(key)
+    })
+  }
+
+  self.nargs = function (key: string | string[] | Dictionary<number>, value?: number) {
+    argsert('<string|object|array> [number]', [key, value], arguments.length)
+    populateParserHintSingleValueDictionary(self.nargs, 'narg', key, value)
+    return self
+  }
+
+  self.choices = function (key: string | string[] | Dictionary<string | string[]>, value?: string | string[]) {
+    argsert('<object|string|array> [string|array]', [key, value], arguments.length)
+    populateParserHintArrayDictionnary(self.choices, 'choices', key, value)
```

Comment:
Misspelling `Dictionnary` a few times, should be `Dictionary`.

Reply:
Fixed.